dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorIMM.java

```diff
@@ -24,98 +25,102 @@
         load.addTags(Tag.STARTLOAD);
 
         return eventSequence(
-				load,
-				super.visitStart(e),
-				newJump(expressions.makeNEQ(resultRegister, one), (Label) e.getThread().getExit())
+                load,
+                super.visitStart(e),
+                newJump(expressions.makeNEQ(resultRegister, one), (Label) e.getThread().getExit())
         );
-	}
+    }
 
-	// =============================================================================================
+    // =============================================================================================
     // =========================================== C11 =============================================
     // =============================================================================================
 
-	@Override
-	public List<Event> visitAtomicCmpXchg(AtomicCmpXchg e) {
-		Register resultRegister = e.getResultRegister();
+    @Override
+    public List<Event> visitAtomicCmpXchg(AtomicCmpXchg e) {
+        Register resultRegister = e.getResultRegister();
         Expression address = e.getAddress();
-		String mo = e.getMo();
-		Fence optionalFenceLoad = mo.equals(Tag.C11.MO_SC) ? newFence(Tag.C11.MO_SC) : null;
-		Fence optionalFenceStore = mo.equals(Tag.C11.MO_SC) ? newFence(Tag.C11.MO_SC) : null;
+        String mo = e.getMo();
+        Fence optionalFenceLoad = mo.equals(Tag.C11.MO_SC) ? newFence(Tag.C11.MO_SC) : null;
+        Fence optionalFenceStore = mo.equals(Tag.C11.MO_SC) ? newFence(Tag.C11.MO_SC) : null;
         Expression expectedAddr = e.getExpectedAddr();
         IntegerType type = resultRegister.getType();
         Expression one = expressions.makeOne(type);
 
         Register regExpected = e.getThread().newRegister(type);
         Register regValue = e.getThread().newRegister(type);
-        Load loadExpected = newLoad(regExpected, expectedAddr, "");
+        Load loadExpected = newLoad(regExpected, expectedAddr);
         loadExpected.addTags(Tag.IMM.CASDEPORIGIN);
-        Store storeExpected = newStore(expectedAddr, regValue, "");
+        Store storeExpected = newStore(expectedAddr, regValue);
         Label casFail = newLabel("CAS_fail");
         Label casEnd = newLabel("CAS_end");
         Local casCmpResult = newLocal(resultRegister, expressions.makeEQ(regValue, regExpected));
         CondJump branchOnCasCmpResult = newJump(expressions.makeNEQ(resultRegister, one), casFail);
         CondJump gotoCasEnd = newGoto(casEnd);
-        Load loadValue = newRMWLoad(regValue, address, extractLoadMo(mo));
-        Store storeValue = newRMWStore(loadValue, address, e.getMemValue(), extractStoreMo(mo));
+        Load loadValue = newRMWLoadWithMo(regValue, address, extractLoadMo(mo));
+        Store storeValue = newRMWStoreWithMo(loadValue, address, e.getMemValue(), extractStoreMo(mo));
 
         return eventSequence(
                 // Indentation shows the branching structure
```

Comment:
If you applied automatic formatting, this note is probably outdated. This recently cause problems to Rene and I was always avoid automatic formatting in these files exactly for this. I would fully remove such comment.

Reply:
Yeah, I did automatic formatting. I forgot about the indentations there. I will add them back in unless you are fine without the indentations.